File: advertisingonrails/app/models/campaign.rb

class Campaign < ApplicationRecord
  enum Objective: [:Awareness, :Reach, :Traffic, :Engagement, :Leads, :Sales, :AppInstalls, :VideoViews]
  enum Status: [:Draft, :Active, :Paused, :Completed, :Cancelled]


  composed_of :money,
    class_name: "Money",
    mapping: [
      ${$mapping}, 
      %w[money_currency currency]
    ]

  composed_of :dateRange,
    class_name: "DateRange",
    mapping: [
      ${$mapping}, 
      %w[dateRange_endDate endDate]
    ]

  has_many :AdAccount, class_name: 'AdAccount'
  has_many :LineItems, class_name: 'LineItem'
  has_many :Kpis, class_name: 'KPI'
  has_many :TrackingPixels, class_name: 'TrackingPixel'
  has_many :Audiences, class_name: 'AudienceSegment'
  has_many :Reports, class_name: 'Report'
  has_many :InsertionOrder, class_name: 'InsertionOrder'

end

File: advertisingonaspdotnet/Domain/Campaign.cs

using advertisingonaspdotnet.Contracts;

namespace advertisingonaspdotnet.Domain;

public class Campaign
{
    public Guid Id { get; set; } = Guid.NewGuid();

 public virtual long? CampaignId { get; set; } 
 public virtual string? Name { get; set; } 
 public virtual Money? TotalBudget { get; set; } 
 public virtual DateRange? Flight { get; set; } 
public virtual AdAccount? AdAccount { get; set; } 
public virtual ICollection<LineItem> LineItems { get; set; } = new List<LineItem>();
public virtual ICollection<KPI> Kpis { get; set; } = new List<KPI>();
public virtual ICollection<TrackingPixel> TrackingPixels { get; set; } = new List<TrackingPixel>();
public virtual ICollection<AudienceSegment> Audiences { get; set; } = new List<AudienceSegment>();
public virtual ICollection<Report> Reports { get; set; } = new List<Report>();
public virtual InsertionOrder? InsertionOrder { get; set; } 
 public virtual ObjectiveType? Objective { get; set; } 
 public virtual CampaignStatus? Status { get; set; } 

    public static Campaign FromRequest(CampaignRequest request) {
        return new Campaign {
            Id = request.Id,
            Name = request.Name,
            TotalBudget = request.TotalBudget,
            Flight = request.Flight,
            Objective = request.Objective,
            Status = request.Status,
        };
    }
}
